frontend/lib/resolution/default-functions.cpp

```diff
@@ -143,6 +152,73 @@ areOverloadsPresentInDefiningScope(Context* context,
   return false;
 }
 
+static bool
+areOperatorOverloadsPresentInDefiningScope(Context* context,
+                                           const QualifiedType& typeForScope,
+                                           const QualifiedType& lhsType,
+                                           const QualifiedType& rhsType,
+                                           UniqueString name) {
+
+  CHPL_ASSERT(lhsType.type());
+  auto ids = getMatchingIdsInDefiningScope(context, typeForScope.type(), name);
+
+  // nothing found
+  if (ids.numIds() == 0) return false;
+
+  for (int i = 0; i < ids.numIds(); i++) {
+    auto& idAndFlags = ids.idAndFlags(i);
+
+    // types are not operators
+    if (idAndFlags.isType()) continue;
+
+    // fields are not operators
+    if (idAndFlags.isMethodOrField() && !idAndFlags.isMethod()) continue;
+
+    // Actually simulate a call to the function to find generic but applicable
+    // instantiations and evaluate where clauses.
+    ResolutionContext rcval(context);
+    auto sig = typedSignatureInitialForId(&rcval, idAndFlags.id());
+    std::vector<CallInfoActual> actuals;
+    actuals.push_back({ lhsType, UniqueString() });
+    actuals.push_back({ rhsType, UniqueString() });
+    auto ci = CallInfo(name,
+                       QualifiedType(),
+                       /* isMethodCall */ false,
+                       /* hasQuestionArg */ false,
+                       /* isParenless */ false,
+                       std::move(actuals));
+    CHPL_ASSERT(ci.isOpCall());
+
+    // If the function doesn't apply, this isn't a relevant operator overload.
+    auto faMap = FormalActualMap(sig, ci);
+    if (!isInitialTypedSignatureApplicable(context, sig, faMap, ci).success()) {
+      continue;
+    }
+
+    // If this function was generic, and we couldn't instantiate it (e.g.
+    // due to a 'where' clause), it's not an applicable candidate.
+    if (sig->needsInstantiation()) {
+      // note: this is pretty much the body of doIsCandidateApplicableInstantiating
+      // in resolution queries, but that function is private, and it seemed
+      // not worth it to expose it in the header.
+
+      auto result = instantiateSignature(&rcval, sig, ci, /* poiScope */ nullptr);
+
+      // function didn't apply after instantiation
+      if (!result.success()) continue;
+
+      // 'where' clause evaluated to false, so this doesn't apply
+      if (result.candidate()->whereClauseResult() == TypedFnSignature::WHERE_FALSE)
```

Comment:
Seems like maybe we could collapse this down into e.g., `result.canBeUsed()` or `result.applies()` or something.

Reply:
I'm going to sit on this because there are only two places this occurs, and I'm nervous about `.success()` vs `.canBeUsed()` -- introducing various synonymous-ish things that aren't quite the same.

Reply:
OK.